### PR TITLE
Project: RoR Clubhouse

### PR DESCRIPTION
**Pull request by crgc**
- Added models for Article and User
- Restricted edit access to signed-in users only
- Added a simple form to create new articles
- Linked the current signed-in user to the created article
- Author names are only visible to signed-in users

### CODE REVIEW

File: app/views/articles/_article.html.erb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        You need to sign up or login to see the author
+      </div>
+    <% end %>
+    <div class="card-body">
+      <p class="card-text">
+        <%= article.article %>
+      </p>
+      <%= link_to article, class: 'btn btn-secondary text-white' do %>
+        <i class="bi bi-link-45deg"></i>
+      <% end %>
```
Comment (acushlakoncept) [inline app/views/articles/_article.html.erb]: There should be absolutely NO logic in views

File: app/views/articles/show.html.erb [hunk 2]
```diff
@@ -1,9 +1,28 @@
...
+  <div class="card-body">
+    <p class="card-text">
+      <%= @article.article %>
+    </p>
+    <%= link_to articles_path, class: 'btn btn-secondary text-white' do %>
+      <i class="bi bi-arrow-left-square"></i>
+    <% end %>
+    <%= link_to edit_article_path(@article), class: 'btn btn-secondary text-white' do %>
+      <i class="bi bi-pencil"></i>
+    <% end %>
```
Comment (acushlakoncept) [inline app/views/articles/show.html.erb]: There should be absolutely NO logic in views

**Review by acushlakoncept — CHANGES_REQUESTED** [review]
# Status: Changes Required ♻️

Hello @chuckbuckethead and @Diegodsha, Great job! 👏  Both of you have done well on this project.

* Well written ReadMe file 👍 💯 
* 

Let's make a few changes, and we will be done.

- [x] Lets maintain a RESTful route by allowing our application to expose only the routes we want. Example `resources :articles, only: [:new, :create, :index]` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_routes_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))

- [x] I should not be able to create empty article, kindly add validation to it. 

- [x] There should be absolutely NO logic in views. Views are only for displaying information that became available to it from its corresponding controller not generating it.  ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright)
I have gone ahead and made some comments on your codebase see affected files below:

**Don't know where to start or how to?** See  [Rails View Helpers Best Practices](https://medium.com/@acushla4real/rails-view-helpers-best-practices-guide-d7d7b6db3db4) 

- [x] Do not forget to add `yarn` to your getting started instructions, perfect after the `bundle install`. I couldn't run the project without running `yarn`


### Cheers and Happy coding! 💻 👏

(Feel free to contact me on slack(@acushla) or reply to this thread, if you have any additional question)
Uduak Essien
[Twitter](https://twitter.com/acushlakoncept) | [LinkedIn](https://www.linkedin.com/in/acushlakoncept/)

File: app/views/articles/_article.html.erb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  <div class="card text-center">
+    <% if user_signed_in? %>
+      <div class="card-header">
+        Written by: <%= username(article.user.email) %>
+      </div>
+    <% else %>
+      <div class="card-header">
+        You need to sign up or login to see the author
+      </div>
+    <% end %>
```
Comment (bini-i) [inline app/views/articles/_article.html.erb]: Remove logic from views:
you can put the whole if statement in a method called show_artilce under articles helpers
```ruby
def show_artilce(article) 
      if user_signed_in? %>
          render html: tag.div "Written by: ${username(article.user.email})"
     else
         render html: tag.div "You need to sign up or login to see the author"
     end
end
```
call show article method from your view,
Please do the same for all logics you have in your views

**Review by bini-i — CHANGES_REQUESTED** [review]
# Status: Changes Required :yellow_circle:        
### Hello team :wave: 

Thanks for working on the changes required by the previous TSE but you have to work on all the requirements for this project to be approved.

- [x] Please make sure your routes to article resources should for only new, create and index actions. kindly  modify your routes.rb file with `resources :articles, only: [:new, :create, :index]`
image
- [x] Kindly add validations to your Article model so that it does not allow empty article creation.
example: `validates :article, presence: true` , this makes sure article attribute has value before it is saved. You can refer [here](https://guides.rubyonrails.org/active_record_validations.html) for more validations.

- [x] Remove all the logics from your views. You can put them in a helper method and call the method from your views. Refer inline comment below

#### If you have any question kindly reach me on slack @Binyam Hailemeskel

### HAPPY CODING :computer: :blush: 

> #### Reviewed by Binyam Hailemeskel (Part-Time TSE)
> GitHub: [@bini-i](https://github.com/bini-i) 
> Twitter: [@binyamshewa](https://twitter.com/binyamshewa)
> LinkedIn: [LinkedIn](https://www.linkedin.com/in/binyam-hailemeskel-728048151/)

**Review by NtwaliHeritier — APPROVED** [review]
### **STATUS: APPROVED✔✔✔✔**

Hello team,
Great implementation of the project.
You can merge and move on👍👌

Happy coding💻💻💻

**Pull request merged**